src/HasMedia/HasMediaCollectionsTrait.php

```diff
@@ -2,300 +2,48 @@
-			if ( $model->shouldAutoProcessMedia() ) {
-				$model->processMedia( collect( request()->only( $model->getMediaCollections()->map->getName()->toArray() ) ) );
-			}
-		} );
-	}
-
-	protected function shouldAutoProcessMedia() {
-		if ( config( 'media-collections.auto_process' ) && property_exists( $this,
-				'autoProcessMedia' ) && ! ! $this->autoProcessMedia ) {
-			return true;
-		}
-
-		return false;
-	}
-
-	protected function initMediaCollections() {
-
-		$this->mediaCollections = collect();
-
-		$this->registerMediaCollections();
-
-//        V  RegisterAllMediaConversions sa to vola zas
-	}
-
-//    public function registerAllMediaConversions(Media $media = null)
-//    {
-//        if(empty($this->mediaCollections)){
-//            $this->registerMediaCollections();
-//        }
-//
-//        collect($this->mediaCollections)->each(function (MediaCollection $mediaCollection) use ($media) {
-//            $actualMediaConversions = $this->mediaConversions;
-//
-//            $this->mediaConversions = [];
-//
-//            ($mediaCollection->mediaConversionRegistrations)($media);
-//
-//            $preparedMediaConversions = collect($this->mediaConversions)
-//                ->each(function (Conversion $conversion) use ($mediaCollection) {
-//                    $conversion->performOnCollections($mediaCollection->name);
-//                })
-//                ->values()
-//                ->toArray();
-//
-//            $this->mediaConversions = array_merge($actualMediaConversions, $preparedMediaConversions);
-//
-//        });
-//
-//        $this->registerMediaConversions($media);
-//
-//    }
-
-	/**
-	 * Register new Media Collection
-	 *
-	 * Adds new collection to model and set its name.
-	 *
-	 * @param $name
-	 *
-	 * @return MediaCollection
-	 * @throws MediaCollectionAlreadyDefined
-	 */
-	public function addMediaCollection( $name ): MediaCollection {
-
-//		if ($this->mediaCollections->has($name)) {
-//
-//		    throw new MediaCollectionAlreadyDefined("Error in $name");
-//		}
-
-//		V starej verzii to zbehlo 6 krat, teraz 12. Vola to aj FileAdder, vola to aj register Conversions,
-// pokazde sa to vola a preto asi nema zmysel chceckovat ci este nie su zaregistrovane. Nema to podla mna ani zmysel, vzdy sa len prepise kolekcia.
-
-		$collection = MediaCollection::create( $name );
-
-        $this->mediaCollections->put( $name, $collection );
-
-		return $collection;
-	}
-
-	/**
-	 * Returns a collection of Media Collections
-	 *
-	 * @return Collection|MediaCollection[]
-	 */
-	public function getMediaCollections(): Collection {
-		return $this->mediaCollections;
-	}
-
-	/**
-	 * Returns a Media Collection according to the name
-	 *
-	 * If Media Collection was not registered on this model, null is returned
-	 *
-	 * @param $name
-	 *
-	 * @return MediaCollection|null
-	 */
-	public function getMediaCollection( $name ): ?MediaCollection {
-
-		return $this->mediaCollections->get( $name );
-	}
+trait HasMediaCollectionsTrait
```

Comment:
Okay, if we answer two questions below, we may end up removing this file entirely.

Reply:
If we remove the features mentioned later, we can.